### ISSUE TITLE
[BUG] Classes not detected inside objects with computed property (using `clsx` callee)

### ISSUE THREAD

**Issue by redocean10** [francoismassart/eslint-plugin-tailwindcss#331]
**Describe the bug**
The plugin won't identify classes inside objects that have a computed property (`object[property]` or `object.property`), so doesn't check for any rules (`classnames-order` in this case).

**To Reproduce**
```cjs
// .eslintrc.cjs
module.exports = {
  extends: [
    "plugin:tailwindcss/recommended",
  ],
  settings: {
    tailwindcss: {
      callees: ["clsx"],
    }
  },
  rules: {
    "tailwindcss/no-custom-classname": "off",
     },
  }
```

```jsx
// Alert.jsx
import clsx from 'clsx'

let type = 'error'
const alertClass = clsx(
  {
    success: 'border-green-400 bg-green-100 text-green-700',
    warning: 'text-yellow-700 border-yellow-400 bg-yellow-100',
    error: 'bg-red-100 border-red-400 text-red-700',
  }[type],
)
```

**Expected behavior**
The plugin should detect the classes and warn of the invalid order.

**Screenshots**
With the computed property (no linting):
![image](https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/127218368/dc42c2fa-5313-4d94-a9b3-519d22972e5a)

Without property (linting works fine):
![image](https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/127218368/bdb3c9ae-3bad-47bb-a393-34be4b219d07)

**Environment (please complete the following information):**
 - OS: Linux Mint 21.2 x86_64 
 - Softwares + version used:
     -  VSCode 1.87.2
     -  `"eslint-plugin-tailwindcss": "3.15.1"`

**Current fix/ workaround:**

I managed to fix this issue for the classnames-order rule, by updating the `callExpressionVisitor` function inside classnames-order.js:

```javascript
const callExpressionVisitor = function (node) {
  const calleeStr = astUtil.calleeToString(node.callee);
  if (callees.findIndex((name) => calleeStr === name) === -1) {
    return;
  }

  node.arguments.forEach((arg) => {
    if (arg.type === 'MemberExpression' && arg.object.type === 'ObjectExpression') {
      arg.object.properties.forEach((prop) => {
        if (prop.value.type === 'Literal') {
          sortNodeArgumentValue(node, prop.value);
        }
      });
    } else {
      sortNodeArgumentValue(node, arg);
    }
  });
};
```

**Comment by kachkaev** [comment]
Hmm not sure if creating an object and then picking a single element from it is an efficient way of achieving what you are after. If there is ever a fix, it will encourage odd-looking `clsx` calls which may look confusing for collaborators. In the meantime, you can do this:

```ts
const alertClass = clsx(
  type === 'success' && 'border-green-400 bg-green-100 text-green-700',
  type === 'warning' && 'text-yellow-700 border-yellow-400 bg-yellow-100',
  type === 'error' && 'bg-red-100 border-red-400 text-red-700',
)
```

Or, even better, you can leverage [`cva`](https://www.npmjs.com/package/class-variance-authority):

```ts
import cva from 'class-variance-authority';

const alertVariants = cva('', {
  variants: {
    type: {
      success: 'border-green-400 bg-green-100 text-green-700',
      warning: 'text-yellow-700 border-yellow-400 bg-yellow-100',
      error: 'bg-red-100 border-red-400 text-red-700',
    }
  }
});

const alertClass = cva({ type });
```

Both alternatives are covered by this plugin.
